File: src/algorithms/algorithmFactory.js
import Bfs from './bfs.js'
import Dfs from './dfs.js'

class AlgorithmFactory {
	createAlgorithm(algorithm, graph, start, finish){
		let algo = null;
		if (algorithm === 'bfs'){
			algo = new Bfs(graph, start, finish)
		} else if (algorithm === 'dfs'){
			algo = new Dfs(graph, start, finish)
		}
		return algo;
	}
}

export default AlgorithmFactory

File: Algorithms/AlgorithmFactory.cpp
#include "AlgorithmFactory.h"

AlgorithmFactory::AlgorithmFactory() {

}

Algorithms* AlgorithmFactory::switchAlgo(QString str) {
    std::string algoStr = str.toStdString();
    Algorithms* algo;
    if (algoStr == "BFS") {
        std::cout << "BFS selected " << std::endl;
        algo = new BFS(); 
    } else if (algoStr == "DFS") {
        std::cout << "DFS selected " << std::endl;
        algo = new DFS();
    } else if (algoStr == "Random") {
        std::cout << "Random selected" << std::endl;
        algo = new Random();
    }
    return algo;
}


